packages/tasit-action/src/TasitAction.test.js

```diff
@@ -328,21 +328,66 @@ describe("TasitAction.Contract", () => {
       expect(txSubscription.subscribedEventNames()).to.be.empty;
     });
 
-    it("should emit error event when orphan/uncle block occurs", async () => {
-      txSubscription = simpleStorage.setValue("hello world");
-
+    // Note: Block reorganization is the situation where a client discovers a
+    //  new difficultywise-longest well-formed blockchain which excludes one or more blocks that
+    //  the client previously thought were part of the difficultywise-longest well-formed blockchain.
+    //  These excluded blocks become orphans.
+    it("should emit error event when block reorganization occurs - block excluded", async () => {
       const confirmationFn = sinon.fake();
       const errorFn = sinon.fake();
 
       const confirmationListener = message => {
         confirmationFn();
       };
 
+      const errorListener = message => {
+        const { error, eventName } = message;
+
+        // Note: This assertion will not fail the test case (UnhandledPromiseRejectionWarning)
+        expect(error.message).to.equal(
+          "Your message has been included in an uncle block."
+        );
+
+        // But asserting fake function, if that throws, test case will fail.
+        errorFn();
+      };
+
+      const snapshotId = await createSnapshot(provider);
+
+      txSubscription = simpleStorage.setValue("hello world");
+
       txSubscription.on("confirmation", confirmationListener);
```

Comment:
I think whenever we can we'll want to set it up so users can add the confirmation listener before sending the action

Reply:
Extracted to: https://github.com/tasitlabs/TasitSDK/issues/125

Reply:
Perfect - resolving this